cfpb/cfgov-refresh
# Remove browserSync and gulp watch

`browserSync` is in dependencies because it's a dependency in production as part of the gulp workflow, but it is not run. browserSync also has snyk reported vulnerabilities https://snyk.io/test/npm/browser-sync. ~~This PR sets up an architecture to keep it in devDependencies.~~

## Additions
- ~~Adds `browser-sync` utility for running browser-sync dependency in non-production environments.~~

## Removals

- ~~Removes `browsersync` gulp task and merges it with the `watch` task.~~
- browser-sync dependency.
- gulp watch task.

## Changes

- Sets the `NODE_ENV` value in `cli_flags.sh`

## Testing

1. Run `./frontend.sh` to get into development state.
2. ~~Run `gulp watch` and see that it runs.~~
3. ~~Edit a CSS file (like `main.less`) and save it and see that it re-compiles.~~
4. ~~Edit a JS file (like `common.js`) and save it and see that it re-compiles.~~
5. ~~ctrl+c to quit watch.~~
1. Run `./frontend.sh production` to get into production state.
2. ~~Run `gulp watch` and see that it says `NODE_ENV` is in `production`.~~
3. Run `gulp build` and see that it passes.

## Todo

- ~~If you run `./frontend.sh production` first it modifies package-lock.json, and when `./frontend.sh` is run subsequently, the devDependencies aren't installed—requiring `npm install` be run between the two. This PR should not be merged before this is resolved.~~
